File: java/dsa/data-structures/stack-queue/implement-stack-using-queues.java
class MyStack {
    private Queue<Integer> qStorage = new LinkedList<>();

    public void push(int x) { // @viz: push_start
        qStorage.offer(x); // @viz: push_val
        for (int i = 0; i < qStorage.size() - 1; i++) { // @viz: loop @viz: push_end
            qStorage.offer(qStorage.poll()); // @viz: rotate_pop @viz: rotate_push
        }
    }

    public int pop() { // @viz: pop_start
        return qStorage.poll(); // @viz: pop_val @viz: pop_end
    }

    public int top() { // @viz: top_start
        return qStorage.peek(); // @viz: top_val @viz: top_end
    }

    public boolean empty() { // @viz: empty_start
        return qStorage.isEmpty(); // @viz: empty_check
    }
}

File: cpp/dsa/data-structures/stack-queue/implement-stack-using-queues.cpp
class MyStack {
private:
    queue<int> q;

public:
    void push(int x) { // @viz: push_start
        q.push(x); // @viz: push_val
        for (int i = 0; i < q.size() - 1; i++) { // @viz: loop @viz: push_end
            int val = q.front(); // @viz: rotate_pop
            q.pop();
            q.push(val); // @viz: rotate_push
        }
    }

    int pop() { // @viz: pop_start
        int val = q.front(); // @viz: pop_val
        q.pop();
        return val; // @viz: pop_end
    }

    int top() { // @viz: top_start
        return q.front(); // @viz: top_val @viz: top_end
    }

    bool empty() { // @viz: empty_start
        return q.empty(); // @viz: empty_check
    }
};